packages/communication-react/review/beta/communication-react.api.md

```diff
@@ -1737,20 +1703,16 @@ export interface ChatAdapterThreadManagement {
     loadPreviousChatMessages(messagesToLoad: number): Promise<boolean>;
     removeParticipant(userId: string): Promise<void>;
     removeResourceFromCache(resourceDetails: ResourceDetails): void;
-    sendMessage(content: string, options?: SendMessageOptions): Promise<void>;
-    sendMessageWithAttachments(content: string, attachments: AttachmentMetadata[]): Promise<void>;
+    sendMessage(content: string, options?: SendMessageOptions | /* @conditional-compile-remove(attachment-upload) */ MessageOptions): Promise<void>;
     sendReadReceipt(chatMessageId: string): Promise<void>;
     sendTypingIndicator(): Promise<void>;
     setTopic(topicName: string): Promise<void>;
-    updateMessage(messageId: string, content: string, metadata?: Record<string, string>, options?: {
-        attachmentMetadata?: AttachmentMetadata[];
-    }): Promise<void>;
+    updateMessage(messageId: string, content: string, metadata?: Record<string, string>, options?: MessageOptions): Promise<void>;
```

Comment:
Shouldn't this be a union of the types?
```suggestion
    updateMessage(messageId: string, content: string, options?: Record<string, string> | MessageOptions): Promise<void>;
```

Reply:
Also, we are deprecating the metadata into options, and we can't do this if we change the number of parameters in the method

Reply:
updated